### PR TITLE
[FRONT-802] Add connect wallet modal

### PR DESCRIPTION
**Pull request by its-edmund**

### CODE REVIEW

File: src/components/WalletModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+    return (
+        <div>
+            {button ? (
+                React.cloneElement(button, { onClick: openModal })
+            ) : (
+                <button onClick={openModal} className="button">
+                    Connect a Wallet
+                </button>
+            )}
```
Comment (mondoreale) [inline src/components/WalletModal/index.tsx:95]: We can actually go with just the button. No need to prepare it for different use cases. #kiss

File: src/components/WalletModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        font-weight: bold;
+    }
+
+    .walletOption:hover {
+        background-color: #dfe3e8;
+    }
+`
+
+const UnstyledWalletModal = ({ button }: { button?: ReactElement }) => {
+    const [modalIsOpen, setModalIsOpen] = useState(false)
```
Comment (mondoreale) [inline src/components/WalletModal/index.tsx:77]: `useState<boolean>(false)`

File: src/components/WalletModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                </button>
+            )}
+            <ReactModal
+                isOpen={modalIsOpen}
+                contentLabel="Connect a wallet"
+                style={customStyles}
+                onRequestClose={closeModal}
+            >
+                <StyledModalContent>
+                    <div className="contentContainer">
```
Comment (mondoreale) [inline src/components/WalletModal/index.tsx:103]: Let's convert them class names to styled components.

**Review by mondoreale — CHANGES_REQUESTED** [review]
A bit similar to the other modal I looked at.

- Let's avoid `className` props written by hand.
- KISS and avoid unnecessary reusability handling.

I'd like to ask you to look up on the web the SVGs for Metamask and Keystone (WalletConnect and Coinbase are in figma). Let's make all of them icons sharp. 🔪 

We're on the right path, sir. 🚀

File: src/components/WalletModal/index.tsx (hunk 1 above)
Comment (mxinburritos) [reply src/components/WalletModal/index.tsx:95]: Gotcha! Thanks for the pointer!

Comment (mondoreale) [reply src/components/WalletModal/index.tsx]: We can also apply this to #11… 🤔

**Comment by mondoreale** [comment]
I'll wait for #11 to get merged first. This one is missing a `package.json` update (no `react-modal`).

**Comment by mondoreale** [comment]
@its-edmund we'll have to get the recent `development` branch in here and make it work with the current functionality of the "Connect a wallet" button. Clicking on MM item in the modal should do whatever the current "Connect a wallet" button does.

**Review by mondoreale — APPROVED** [review]
Looks good. We need to link things up. :)

**Pull request merged**